test/docgen/view/documentation.test.ts

```diff
@@ -182,3 +182,19 @@ describe('csharp', () => {
     expect(markdown.render()).toMatchSnapshot();
   });
 });
+
+test('throws uninstallable error on dependency conflict', async () => {
+  // this package decalres a fixed peerDependency on constructs, which conflicts with its other dependencies
+  return expect(Documentation.forPackage('cdk8s-mongo-sts@0.0.5')).rejects.toThrowError(UnInstallablePackageError);
+});
+
+test('throws uninstallable error on missing spec in dependencies', async () => {
+  // this package has a corrupt package.json that doesn't contain a spec for some dependencies
+  return expect(Documentation.forPackage('cdk-codepipeline-bitbucket-build-result-reporter@0.0.7')).rejects.toThrowError(UnInstallablePackageError);
+});
+
+test('throws corrupt assembly', async () => {
+  const docs = await Documentation.forPackage('@epilot/cdk-constructs@1.0.7');
```

Comment:
Possibly. I don't think it makes that much of a difference since dependencies are still fetched from NPM.

The right thing to do here is to create a fixture project that simulates the scenario under test...but...for another time :)